### PR TITLE
CLN: remove unused try_cast kwarg from ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1318,7 +1318,7 @@ def shift(self, periods, axis=0, mgr=None):
 
         return [self.make_block(new_values)]
 
-    def eval(self, func, other, errors='raise', try_cast=False, mgr=None):
+    def eval(self, func, other, errors='raise', mgr=None):
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py]: Given recent implementation of ops.dispatch_to_series, we only ever get here with `try_cast=False`, so we can remove the kwarg altogether.

**Comment by jbrockmendel** [comment]
Travis failure is in feather

**Comment by jreback** [comment]
this is fine (I think you have these changes in another PR as well), but happy to merge this. ping on green.

**Comment by jreback** [comment]
thanks!

**Pull request merged**